prov/verbs/src/verbs_ep.c

```diff
@@ -464,10 +464,11 @@ vrb_alloc_init_ep(struct fi_info *info, struct vrb_domain *domain,
 		if (vrb_alloc_wrs(ep))
 			goto err1;
 	}
-
+	ofi_mutex_lock(&vrb_info_mutex);
```

Comment:
The reason for needing this lock is obscure given the code.  Is there a better location for this lock, or better way to restructure the code, so that the locking is closer to what it actually protects?

I'm not overly concerned about acquiring this lock multiple times, since it's in the control paths.  It seems like it's the price to pay for handling device removal at any point in the code.

Reply:
The code actually accessing the protected resource (vrb_util_prov.info) is in the util code, where the lock is not always needed (the lock is only needed for verbs). 

I think adding a comment here should be sufficient.

Reply:
I agree with you @shefty -- this lock is a bit obscure. 
We could make the util code aware of this lock by adding a pointer to it in `struct util_prov`:
```
struct util_prov {
        const struct fi_provider        *prov;
        const struct fi_info            *info;
        ofi_mutex_t                     *info_lock;  /* address of the lock to protect the info */
        ofi_alter_info_t                alter_defaults;
        const int                       flags;
};
```
Then the util code can perform the locking if `info_lock` is valid. What do you think?

Reply:
I think that would be better without much complexity.

Reply:
Thanks to both of you for your feedback! I've updated the PR with the suggested changes.